docs/wpf/region-navigation/confirming-navigation.md

```diff
@@ -1,2 +1,55 @@
 # Confirming Navigation
 
+You will often find that you will need to interact with the user during a navigation operation, so that the user can confirm or cancel it. In many applications, for example, the user may try to navigate while in the middle of entering or editing data. In these situations, you may want to ask the user whether he or she wants to save or discard the data that has been entered before continuing to navigate away from the page, or whether the user wants to cancel the navigation operation altogether. Prism supports these scenarios via the **IConfirmNavigationRequest** interface.
+
+The **IConfirmNavigationRequest** interface derives from the **INavigationAware** interface and adds the **ConfirmNavigationRequest** method. By implementing this interface on your view or view model class, you allow them to participate in the navigation sequence in a way that allows them to interact with the user so that the user can confirm or cancel the navigation. One way of displaying a confirmation is to use a simple windows message box. For something more complex, the **Dialog Service**, as described in [Dialog Service](../dialog-service.md) could be used.
+
+The **ConfirmNavigationRequest** method provides two parameters, a reference to the current navigation context as described earlier, and a callback method that you can call when you want navigation to continue.
+
+The following steps summarize the process of confirming navigation using an **InteractionRequest** object:
+
+1. Navigation operation is initiated via a **RequestNavigate** call.
+1. If the view or view model of the current view implements **IConfirmNavigation**, **ConfirmNavigationRequest** is called.
+1. The view displays the confirmation UI and awaits the user's response.
+1. Continuation callback is invoked to continue or cancel the pending navigation operation.
+1. The navigation operation is completed or canceled.
+
+To illustrate this, check out the sample app at [22-ConfirmCancelNavigation](https://github.com/PrismLibrary/Prism-Samples-Wpf/tree/master/22-ConfirmCancelNavigation).
+
+```cs
+public class ViewAViewModel : BindableBase, IConfirmNavigationRequest
+{
+    public ViewAViewModel()
+    {
+    }
+
+    public void ConfirmNavigationRequest(NavigationContext navigationContext, Action<bool> continuationCallback)
+    {
+        bool result = true;
+
+        if (MessageBox.Show("Do you to navigate?", "Navigate?", MessageBoxButton.YesNo) == MessageBoxResult.No)
```

Comment:
maybe put a comment here about not using MessageBox in production code, this is just demo code

Reply:
Hi @brianlagunas , updated!
thanks,
Mike